recipes/mavsdk/all/conanfile.py

```diff
@@ -0,0 +1,167 @@
+from os.path import join
+from conan import ConanFile
+from conan.tools.files import get, copy
+from conan.tools.files import replace_in_file, rmdir
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.scm import Git
+from conan.tools.env import Environment
+
+required_conan_version = ">=1.53.0"
+
+
+class MAVSDKConan(ConanFile):
+    name = "mavsdk"
+    license = "BSD-3-Clause"
+    author = "SINTEF Ocean"
+    homepage = "https://mavsdk.mavlink.io/main/en/"
+    url = "https://github.com/mavlink/MAVSDK.git"
+    description = "C++ library to interface with MAVLink systems"
+    settings = "os", "compiler", "build_type", "arch"
+    package_type = "library"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False]
+        }
+    default_options = {
+        "fPIC": True,
+        "shared": False
+        }
+
+    @property
+    def export_sources(self):
+        pass
+        #export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("jsoncpp/[>=1.9.5 <2]")
+        self.requires("tinyxml2/[>=9.0.0 <10]")
+        self.requires("libcurl/[>=7.86.0 <8]")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def build_requirements(self):
+        pass
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version]["sdk"], strip_root=True)
+        sources = self.conan_data["sources"][self.version]["mavlink"]
+        mavlink_dir = join(self.source_folder, "src", "third_party",
+                           "mavlink", "include", "mavlink", "v2.0")
+        git = Git(self, folder=mavlink_dir)
+        git.fetch_commit(sources["url"], sources["commit"])
+
+        # TODO: create instead patch files to simplify building versions
+        replace_in_file(self, join(self.source_folder, "CMakeLists.txt"),
+                        'message(STATUS "Version: ${VERSION_STR}")',
+                        f'set(VERSION_STR v{self.version})\nmessage(STATUS "Version: ${{VERSION_STR}}")')
+
+        replace_in_file(self, join(self.source_folder, "src/core/connection.h"),
+                        "#include <unordered_set>",
+                        "#include <unordered_set>\n#include <atomic>")
+
+        link_jsonpp = ["src/cmake/unit_tests.cmake",
+                       "src/plugins/mission_raw/CMakeLists.txt",
+                       "src/plugins/mission/CMakeLists.txt"]
+
+        for f in link_jsonpp:
+            replace_in_file(self, join(self.source_folder, f),
+                            "JsonCpp::jsoncpp", "JsonCpp::JsonCpp")
```

Comment:
This can be better fixed by setting this in the `generate` method:

```py
deps = CMakeDeps(self)
deps.set_property("jsoncppp", "cmake_target_name", "JsonCpp::jsoncpp")
deps.generate()
```

This way Conan will change the generated target name to match those used by the project, removing the need to patch the sources themselves!